### PR TITLE
RA - Add Soviet08a

### PR DESCRIPTION
**Pull request by Smittytron**
This is a standard destroy the enemy mission. @Murto did the early triggers and I picked up where he left off.

### CODE REVIEW

**Comment by Muh-Muh** [comment]
Small issue:
If you capture the enemies production facilities some things can not be built (bc. tech centre can't be built) and should be disabled. These are:
Gap generator
Chronosphere
Tanya (x2, I assume the second Tanya is the .autotarget one)
Radar Jammer
Longbow
Cruiser

Other than that I didn't find any issues playing it (so far only on easy) or in the code.

**Comment by Smittytron** [comment]
Updated

**Comment by Muh-Muh** [comment]
Is it possible that you forgot to push? rules.yaml still seems to be missing those actors.

**Comment by Smittytron** [comment]
I didn't put them at the end, I grouped them with their type i.e. structures, vehicles ect.
MRJ, HELI, CA, GAP, PDOX and E7 are all in there.

**Comment by Muh-Muh** [comment]
Oh right. Nevermind then.

File: mods/ra/maps/soviet-08a/soviet08a.lua [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+				scout.Stop()
+				scout.Move(waypoint1.Location, 1)
+			end
+			Trigger.RemoveProximityTrigger(id)
+		end
+	end)
+end
+
+Tick = function()
+	greece.Cash = 1000
```
Comment (abcdefg30) [inline mods/ra/maps/soviet-08a/soviet08a.lua]: This feels like we're hiding some other issue. What exactly was the reason again why the AI to runs out of money? Do the harvesters get stuck?

File: mods/ra/maps/soviet-08a/rules.yaml [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+Player:
+	PlayerResources:
+		DefaultCash: 4000
+
+CAMERA:
+	RevealsShroud:
+		Range: 6c0
+
+LST.Reinforcement:
```
Comment (abcdefg30) [inline mods/ra/maps/soviet-08a/rules.yaml]: Those should be unselectable or have no voice when selecting them.

**Review by abcdefg30 — COMMENTED** [review]
Some of the sandbag walls probably have the wrong owner, as they are crushed by allied tanks.

File: mods/ra/maps/soviet-08a/rules.yaml [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+	Inherits: LST
+	RejectsOrders:
+	-Buildable:
+	RenderSprites:
+		Image: lst
+
+POWERPROXY:
```
[the remark this answers is not in the document]
Comment (Smittytron) [reply mods/ra/maps/soviet-08a/rules.yaml]: This one is the 1:30 paradrop that gives three flamers. The reinforcement and retreat triggers (sans the mammoth) were written by Murto and are apparently true to the original.

File: mods/ra/maps/soviet-08a/soviet08a.lua (hunk 1 above)
Comment (Smittytron) [reply mods/ra/maps/soviet-08a/soviet08a.lua]: I just figured there's no way AI can keep up constant infantry and vehicle production off of one harvester mining long-distance. That takes four in competitive play.

The mission Evacuation does Cash = 1000 as well.

**Comment by Smittytron** [comment]
Updated

File: mods/ra/maps/soviet-08a/soviet08a.lua (hunk 1 above)
Comment (MunWolf) [reply mods/ra/maps/soviet-08a/soviet08a.lua]: I really don't like cheating AI, but I guess if we didn't this mission would be too easy :/

**Comment by MustaphaTR** [comment]
Mechanic and Helipad are available if you capture the allied tech.

**Comment by Smittytron** [comment]
Updated

File: mods/ra/maps/soviet-08a/rules.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (Smittytron) [reply mods/ra/maps/soviet-08a/rules.yaml]: Renamed to `scripteddrop` for clarity.

File: mods/ra/maps/soviet-08a/soviet08a-AI.lua [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+	hard = DateTime.Seconds(12),
+	tough = DateTime.Seconds(5)
+}
+
+AlliedInfantry = { "e1", "e3" }
+AlliedVehiclesUpgradeDelay = DateTime.Minutes(10)
+AlliedVehicleType = "Normal"
+AlliedVehicles =
+{
+	Normal = { "1tnk", "2tnk" },
```
Comment (GraionDilach) [inline mods/ra/maps/soviet-08a/soviet08a-AI.lua]: Wouldn't it make sense to produce jeeps as well at this point?

File: mods/ra/maps/soviet-08a/soviet08a.lua (hunk 1 above)
Comment (Mailaender) [reply mods/ra/maps/soviet-08a/soviet08a.lua]: The map doesn't seem to have ore mines so the trucks run low soon unless you expand.

**Comment by Smittytron** [comment]
Rebased

File: mods/ra/maps/soviet-08a/soviet08a.lua [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+			Media.PlaySpeechNotification(ussr, "MissionFailed")
+		end)
+	end)
+	Trigger.OnPlayerWon(ussr, function()
+		Trigger.AfterDelay(DateTime.Seconds(1), function()
+			Media.PlaySpeechNotification(ussr, "MissionAccomplished")
+		end)
+	end)
+
+	Harvester.FindResources()
```
Comment (abcdefg30) [inline mods/ra/maps/soviet-08a/soviet08a.lua]: This loos odd. Doesn't the harvester start mining on its own?

Comment (Smittytron) [reply mods/ra/maps/soviet-08a/soviet08a.lua]: The refinery spawned one will. This is to get the one at 57,56 going.

**Comment by Smittytron** [comment]
Updated: Tested this without bounties and saw that I all of a sudden couldn't make it off the beachhead on highest difficulty. Rebalanced to the point where I could beat it again with an AI delay and added a money crate to the church.

**Comment by reaperrr** [comment]
Moving to Next+1 milestone so it's position in the PR queue won't make it miss another release.

**Review by Orb370 — DISMISSED** [review]
Played through the mission on normal difficulty. No issues.

**Comment by pchote** [comment]
@Smittytron are you able to look into the issue that @SoScared raised? Do we have any other blockers for getting this merged (aside from needing a rebase)?

**Comment by Smittytron** [comment]
> @Smittytron are you able to look into the issue that @SoScared raised? Do we have any other blockers for getting this merged (aside from needing a rebase)?

I don't think there is a way to fix SoScared's point. I assume the transport would get through after the subs move out of the way.

The only potential blocker is @abcdefg30 asked me to look at adding Allied naval attacks to this mission. I've tried both setting up production and having ships come in off the map, but so far I haven't been able to get either method to work. I'll have more time to try to figure it out around the Thanksgiving holiday.

**Comment by Smittytron** [comment]
Updated.
Added two naval patrols. I also removed Real tough guy difficulty and added a gem mine by the allied base.
I added the mine to a) Give the allied harvester more to do, and b) to ensure that if the player has mined the map out after destroying the allied base, they will have a source of income to finish the naval units off with.

**Review by obrakmann — APPROVED** [review]

**Pull request merged**